Jenkinsfile

```diff
@@ -37,7 +41,40 @@ pipeline {
                 script {
                     currentBuild.displayName = "$BUILD_NUMBER: $DOCKER_TAG"
                 }
+                script {
+                    docker.withRegistry(docker_registry_url, ecr_credential) {
+                        // Pull the master images and any previous builds if we're on a different branch
+                        // docker-compose only looks in the local images and doesn't try to pull when building
+                        ['story-server', 'backstop', 'wcag', 'ruby'].each {
+                            // Ignore failures from docker - it's probably an Image Not Found.
+                            // Other errors like out of disk space will cause problems in other commands
+                            try {
+                                docker.image("${docker_registry}/design-system:${it}").pull()
+                                if (env.BRANCH_NAME != 'master') {
+                                        docker.image("${docker_registry}/design-system:dev_${it}_${env.CA_STYLEGUIDE_VERSION_TAG}").pull()
+                                }
+                            } catch (Exception e) {
+                                    echo "Error pulling ${docker_registry}/design-system:dev_${it}_${env.CA_STYLEGUIDE_VERSION_TAG}"
```

Comment:
Does this one also need to re-throw?

Reply:
Not this one. When you run the first build, this error will always trigger because the image doesn't exist. It's possible that there are other errors - so it logs the attempt to make it easier to find things in that case.

Reply:
It'd be cool to provide a different error message for a 404 vs other errors... perhaps something for another PR tho.

Reply:
It looks like this in the logs - so the actual message is provided by the result of docker pull

```
+ docker pull 979633842206.dkr.ecr.eu-west-1.amazonaws.com/design-system:dev_story-server_pr-402_5c4f209
Error response from daemon: manifest for 979633842206.dkr.ecr.eu-west-1.amazonaws.com/design-system:dev_story-server_pr-402_5c4f209 not found: manifest unknown: Requested image not found
[Pipeline] echo
Error pulling 979633842206.dkr.ecr.eu-west-1.amazonaws.com/design-system:dev_story-server_pr-402_5c4f209
```